apps/web/app/(authenticated)/(app)/app/apis/[apiId]/settings/update-ip-whitelist.tsx

```diff
@@ -33,22 +41,34 @@ type Props = {
 };
 
 export const UpdateIpWhitelist: React.FC<Props> = ({ api, workspace }) => {
-  const { pending } = useFormStatus();
-
   const isEnabled = workspace.plan === "enterprise";
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      ipWhitelist: api.ipWhitelist ?? "",
+      apiId: api.id,
+      workspaceId: api.workspaceId,
+    },
+  });
 
-  return (
-    <form
-      action={async (formData: FormData) => {
-        const res = await updateIpWhitelist(formData);
-        if (res.error) {
-          toast.error(res.error.message);
-          return;
-        }
+  const updateIps = trpc.api.updateIpWhitelist.useMutation({
+    onSuccess() {
+      toast.success("Your ip whitelist has been updated!");
+      router.refresh();
+    },
+    onError(err) {
+      console.log(err);
+      toast.error(err.message);
+    },
+  });
+
+  async function onSubmit(values: z.infer<typeof formSchema>) {
+    updateIps.mutateAsync(values);
+  }
+  const router = useRouter();
```

Comment:
same thing, it looks weird that this is below where we use it

Reply:
Moved